editor/symbol_tooltip.cpp

```diff
@@ -0,0 +1,302 @@
+/**************************************************************************/
+/*  symbol_tooltip.cpp                                                    */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "symbol_tooltip.h"
+#include "editor/plugins/script_text_editor.h"
+#include "editor_help.h"
+
+SymbolTooltip::SymbolTooltip(CodeTextEditor* code_editor) : code_editor(code_editor) {
+	// Initialize the tooltip components
```

Comment:
```suggestion
	// Initialize the tooltip components.
```
[Comment style](https://docs.godotengine.org/en/stable/contributing/development/code_style_guidelines.html#comment-style-guide) nitpick, same elsewhere

Reply:
Make sure to apply this to all your comments before marking ready for review

Reply:
> Make sure to apply this to all your comments before marking ready for review

Sorry just to make sure I understand, you are referring to adding a period to the end of any comments?

Reply:
Yes, we enforce punctuation for valid sentences in comments.